src/neural/cuda/network_cuda.cc

```diff
@@ -68,14 +68,17 @@ static size_t getMaxAttentionHeadSize(const LegacyWeights& weights, int N) {
   const size_t encoder_heads = weights.pol_encoder_head_count;
 
   size_t size = N * 64 *
-                std::max(std::max(embedding_op_size, encoder_dff),
-                         std::max(policy_d_model, encoder_d_model));
+      std::max(std::max(embedding_op_size, encoder_dff), policy_d_model);
 
   // size of matmul_qk matrix = encoder_heads_ * Batch * 64 * 64
   const size_t matmul_qk_size = encoder_heads * N * 64 * 64;
   const size_t output_size = N * (64 * 64 + 8 * 24);
-
   size = std::max(size, std::max(matmul_qk_size, output_size));
```

Comment:
output_size goes in a single allocation right? - so since we're going to multiply by 2 down below, maybe it could be halved? (Or maybe just don't bother :P)